conda_smithy/lint_recipe.py

```diff
@@ -341,10 +345,14 @@ def main(recipe_dir, conda_forge=False):
     # stub out cb3 jinja2 functions - they are not important for linting
     #    if we don't stub them out, the ruamel.yaml load fails to interpret them
     #    we can't just use conda-build's api.render functionality, because it would apply selectors
+    CondaBuildConfig = namedtuple('CondaBuildConfig', 'work_dir')
     env.globals.update(dict(compiler=lambda x: x + '_compiler_stub',
                             pin_subpackage=lambda *args, **kwargs: 'subpackage_stub',
                             pin_compatible=lambda *args, **kwargs: 'compatible_pin_stub',
                             cdt=lambda *args, **kwargs: 'cdt_stub',
+                            load_file_regex=lambda *args, **kwargs: \
+                                    load_file_regex(CondaBuildConfig(recipe_dir), \
+                                                    *args, **kwargs),
```

Comment:
I don't like the fact that this will not work in general. For eg: if a work dir is used. Maybe `load_file_regex=lambda *args, **kwargs: defaultdict(lambda : '')`?

cc @msarahan

Reply:
Yes, the actual value here shouldn't matter for the sake of linting.  It does matter for rendering the recipe, and stubbing it out here will potentially not catch errors in data that gets loaded from the file, but that's an acceptable trade-off to keep the linter fast.  I believe that the assumption is that the linter should not need to download the recipe source in order to lint the recipe, so your function may not even work in many cases.  What @isuruf proposes would be a better stub.